segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLogTests.java

```diff
@@ -56,9 +55,11 @@
  * Unit tests for BookKeeperLog. These require that a compiled BookKeeper distribution exists on the local
  * filesystem. It starts up the local sandbox and uses that for testing purposes.
  */
-public class BookKeeperLogTests extends DurableDataLogTestBase {
+public abstract class BookKeeperLogTests extends DurableDataLogTestBase {
     //region Setup, Config and Cleanup
 
+    static final AtomicBoolean SECURE_BK = new AtomicBoolean();
```

Comment:
1. Do not make this `static`
2. Replace this with a `protected abstract boolean` method and override it in each of the two derived classes.

Reply:
This is called from a static method. We need the caller to be static so that we do not recreate BK server for each test. Because of this the method has to be static and non-virtual.

Reply:
Then I suggest we use a different design for this. Let's pass this as an argument to `setupBookKeeper` and let that guy set it internally. I guess it will still be static, but at least it should be `private` and not be mangled with by sub-classes.

Reply:
Fixed.